crates/driver/src/domain/mempools.rs

```diff
@@ -34,8 +34,31 @@ impl Mempools {
             .boxed()
         })));
     }
+
+    /// Defines if the mempools are configured in a way that could lead to
+    /// significant costs in case a settlement fails onchain submission.
+    pub fn high_risk(&self) -> HighRisk {
+        if self.0.iter().any(|mempool| {
+            matches!(
+                mempool.config().kind,
+                infra::mempool::Kind::Public(infra::mempool::HighRisk::Enabled)
+            )
+        }) {
+            HighRisk::Enabled
+        } else {
+            HighRisk::Disabled
+        }
+    }
 }
 
 #[derive(Debug, Error)]
 #[error("no mempools configured, cannot execute settlements")]
 pub struct NoMempools;
+
+/// Defines if the mempools are configured in a way that could lead to
+/// significant costs in case a settlement fails onchain submission.
+#[derive(Debug, Clone, Copy)]
+pub enum HighRisk {
```

Comment:
I'm not a fan of this name. Specifically, from what I understand, the "risk" here encodes whether or not a settlement can revert? In that case, I would not use the same nomenclature as we used in the past but something more specific to what we are encoding here (whether or not the mempool executes reverted transactions or just ignores them).